blob-cache-handler.php

```diff
@@ -6,26 +6,73 @@
 set_include_path(get_include_path() . PATH_SEPARATOR . $path);
 
 use WindowsAzure\Common\ServicesBuilder;
+use WindowsAzure\Blob\Models\CreateBlobOptions;
 
 class PN_Blob_Cache_Handler {
 
-	private $remote_cache_endpoint = null;
+	private $remote_cache_endpoint = NULL;
+	private $content_type = NULL;
+	private $headers_as_json;
 
-	public function __construct( ) {
+	public function __construct( ){
 
 	}
 
-	public function pn_blob_cache_set( $key, $data, $expire, $account_name, $account_key, $container ) {
+	/*
+	* This takes an array of headers and prepares the data as an array of json data objects containing name & value properties
+	* It also looks for the "Content-Type" declared in the header array so we can set that value for the blob
+	*
+	* SETS
+	* $this->content_type
+	* $this->headers_as_json
+	*/
+	public function prepare_headers( $headers ){
+		if( isset( $headers ) ){
+		    foreach( $headers as $x ){
+		        $name_val_pair = explode( ": ", $x );
+		
+		        $header = (object) array( 'name' => trim( $name_val_pair[0] ), 'value' => trim( $name_val_pair[1] ) );
+		
+		        if( $header->name == "Content-Type" ){
```

Comment:
We should compare the lowercase version of `$header->name` to a lowercase  `content-type` to get a safer match. You can use `strtolower()`